lib/plugins/aws/deploy/lib/checkForChanges.js

```diff
@@ -88,22 +88,37 @@ module.exports = {
 
   // Gives the least recent last modify date across all the functions in the service.
   async getFunctionsEarliestLastModifiedDate() {
+    let fetchError;
     const getFunctionResults = this.serverless.service.getAllFunctions().map((funName) => {
       const functionObj = this.serverless.service.getFunction(funName);
       return this.provider
         .request('Lambda', 'getFunction', {
           FunctionName: functionObj.name,
         })
         .then((res) => new Date(res.Configuration.LastModified))
-        .catch(() => new Date(0)); // Function is missing, needs to be deployed
+        .catch((err) => {
+          fetchError = err;
+          return new Date(0);
+        }); // Function is missing, needs to be deployed
     });
 
-    return BbPromise.all(getFunctionResults).then((results) =>
-      results.reduce((currentMin, date) => {
+    return BbPromise.all(getFunctionResults).then((results) => {
+      if (fetchError && fetchError.providerError && fetchError.providerError.statusCode === 403) {
+        this.serverless.cli.log(
+          [
+            'WARNING: Iam policy misconfiguration: unable to get Lambda functions info.',
+            'Deployment will not be skipped even if service files did not change. ',
+            'Forcing deployment...',
+            fetchError.message,
+          ].join('')
+        );
```

Comment:
Let's also add the `Serverless` and `{color: 'orange'}` params to `this.serverless.cli.log` call similarly to: https://github.com/serverless/serverless/blob/b11171c70c8f5597e2644f7ea8ec82a17a9eee29/lib/plugins/aws/lib/checkIfEcrRepositoryExists.js#L19

Reply:
When I put `{ color : 'orange' }` (I comitted the change) I obtain this. Do you know why ?

![serverlessWarning](https://user-images.githubusercontent.com/7863528/109791626-f107ae00-7c12-11eb-8b76-4f7978f8ca14.png)

Reply:
Thanks for your suggestions ! I applied the changes